packages/vats/src/proposals/restart-vats-proposal.js

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable no-await-in-loop */
+// @ts-check
+
+import { Fail } from '@agoric/assert';
+import { deeplyFulfilledObject, makeTracer } from '@agoric/internal';
+import { M, mustMatch } from '@agoric/store';
+import { E, getInterfaceOf } from '@endo/far';
+
+const trace = makeTracer('RV');
+
+const HR = '----------------';
+
+/**
+ * TODO cover each of these collections
+ * - [x] contractKits
+ * - [ ] psmKit
+ * - [x] governedContractKits
+ * - [ ] vatStore
+ */
+
+/**
+ * @param {BootstrapPowers & import('@agoric/inter-protocol/src/proposals/econ-behaviors.js').EconomyBootstrapSpace} space
+ * @param {object} config
+ * @param {{ skip: Array<string>}} config.options
+ */
+export const restartVats = async ({ consume }, { options }) => {
+  console.log(HR);
+  console.log(HR);
+  trace('restartVats start', options);
+  mustMatch(options, harden({ skip: M.arrayOf(M.string()) }));
+
+  trace('awaiting VaultFactorKit as a proxy for "bootstrap done"');
+  await consume.vaultFactoryKit;
+
+  trace('testing restarts');
+  const { contractKits, governedContractKits } = await deeplyFulfilledObject(
+    harden({
+      contractKits: consume.contractKits,
+      governedContractKits: consume.governedContractKits,
+    }),
+  );
+
+  const { instancePrivateArgs } = await consume.diagnostics;
+
+  const failures = [];
+  /**
+   *
+   * @param {string} debugName
+   * @param {Instance} instance
+   * @param {ERef<AdminFacet>} adminFacet
+   */
+  const tryRestart = async (debugName, instance, adminFacet) => {
+    // TODO document that privateArgs cannot contain promises
+    // TODO try making all the contract starts take resolved values
+    const privateArgs = await deeplyFulfilledObject(
+      // @ts-expect-error cast
+      harden(instancePrivateArgs.get(instance) || {}),
+    );
+
+    console.log(HR);
+    console.log(HR);
+    console.log(HR);
+    console.log(HR);
+    trace('tryRestart', debugName, privateArgs);
+
+    if (options.skip.includes(debugName)) {
+      trace('SKIPPED', debugName);
+      return;
+    }
+    try {
+      await E(adminFacet).restartContract(privateArgs);
+      trace('RESTARTED', debugName);
+    } catch (err) {
+      trace('🚨 RESTART FAILED', debugName, err);
+      failures.push(debugName);
+    }
+  };
+
+  // iterate over the two contractKits and use the adminFacet to restartContract
```

Comment:
two? we don't need `psmKit` here too? getting the psmKit stuff into governedContractKits looked like it was still TODO.
(I'm only reviewing this PR for regressions, so it's not critical to landing this, but seems like an omission)

Reply:
this is describing the behavior. the TODO about including psmKits is up top